src/WalletWithSteps.ts

```diff
@@ -0,0 +1,127 @@
+import { Address, createPublicClient, createTransport } from 'viem';
+
+import { Wallet } from '@/clients/contracts';
+import { type OnStepChange } from '@/gondi';
+
+export class WalletWithSteps {
```

Comment:
this seems like an inheritance, what's the limitation on using it? (
```suggestion
export class WalletWithSteps extends Wallet {
    constructor(WaletConstuctorProps & {onStepChange})
```